src/website/shared/templatetags/gh_tags.py

```diff
@@ -20,8 +19,10 @@ def get_gh_username(user: User) -> str | None:
     """
     Return the Github username of a given Auth.User.
     """
-    social_user = SocialUser.objects.get(id=user.id)  # type: ignore
-    social_account: SocialAccount | None = social_user.github_account
+    social_user = User.objects.get(id=user.id)  # type: ignore
+    social_account: SocialAccount | None = (
+        social_user.socialaccount_set.filter(provider="github").first()  # type: ignore
```

Comment:
Can a user have multiple socialaccounts of type GitHub, does it make sense? It look like an integrity issue to me.

Reply:
No, you are right, They would be two different accounts. 

I was doing the `filter`...`first` pattern to deal superusers that don't necessarily need to log in via github. I'm changing those patterns to the following with try except to make this intention more clear:

```
def get_gh_username(user: User) -> str | None:
    """
    Return the Github username of a given Auth.User.
    """
    social_user = User.objects.get(id=user.id)  # type: ignore
    try: 
        extra_data_login: str = (
            social_user.socialaccount_set.get(provider="github")
                .extra_data.get("login")
        )
        return extra_data_login
    except Exception:
        logger.exception("Failed to get GitHub username for user %s.", user)
        return None
```

Reply:
For example: [6602bc8](https://github.com/Nix-Security-WG/nix-security-tracker/pull/159/commits/6602bc892ab83140b894233d73f1f1b42616b2e4)